src/main/java/teammates/common/datatransfer/StudentAttributes.java

```diff
@@ -109,8 +110,31 @@ public StudentAttributes(Student student) {
                                                           : student.getSectionName();
         this.googleId = (student.getGoogleId() == null) ? ""
                                                         : student.getGoogleId();
-        Long keyAsLong = student.getRegistrationKey();
-        this.key = (keyAsLong == null) ? null : Student.getStringKeyForLongKey(keyAsLong);
+        this.key = student.getRegistrationKey();
+        /*
+         * TODO: this is for backward compatibility with old system.
+         * Old system considers "" as unregistered.
+         * It should be changed to consider null as unregistered.
+         */
```

Comment:
I don't know if it's still relevant or what it really means. I'll just remove it